pkg/generate/app/cmd/newapp.go

```diff
@@ -531,6 +531,9 @@ func (c *AppConfig) run(out io.Writer, acceptors app.Acceptors, components app.C
 		if err != nil {
 			return nil, fmt.Errorf("can't setup %q: %v", p.From, err)
 		}
+		if p.Image != nil && p.Image.HasEmptyDir {
+			fmt.Printf("Image %s uses an EmptyDir volume. Data will not be persistent.\n", p.Image.Name)
```

Comment:
I would say something like `NOTICE: Image X uses an EmptyDir volume.  Data in EmptyDir volumes is not persisted across deployments.`